src/org/deepsymmetry/bytefield/core.cljs

```diff
@@ -531,57 +531,85 @@
+         (draw-box label [{:span (- boxes column)} box-above-style (if (nil? fill) {} {:fill fill})]) ; And there is room for it on the current line.
          (do ; The label doesn't fit on the current line.
-           (draw-box nil [{:span (- boxes column)} box-above-style]) ; Finish off current line with emptiness.
+           (draw-box nil [{:span (- boxes column)} box-above-style (if (nil? fill) {} {:fill fill})]) ; Finish off current line with emptiness.
            (auto-advance-row nil)
-           (draw-box label [{:span boxes :borders #{:left :right}}]))) ; Put the label on its own line.
+           (draw-box label [{:span boxes :borders #{:left :right}} (if (nil? fill) {} {:fill fill})]))) ; Put the label on its own line.
        ;; We are not supposed to draw a label, so just finish the current line if needed.
        (when (not= column boxes)
-         (draw-box nil [{:span (- boxes column)} :box-above]))) ; Finish off current line with emptiness.
+         (draw-box nil [{:span (- boxes column)} :box-above (if (nil? fill) {} {:fill fill})]))) ; Finish off current line with emptiness.
 
      ;; Move on to a new row to draw the gap.
      (auto-advance-row nil)
      (let [y      (diagram-height)
            top    (+ y edge)
            left   @('left-margin @*globals*)
            right  (+ left (diagram-width))
-           bottom (+ y (- height edge))]
-       (draw-line left y left top)
-       (draw-line right y right top)
-       (draw-line left top right (- bottom gap) gap-style)
-       (draw-line right y right (- bottom gap))
-       (draw-line left (+ top gap) right bottom gap-style)
-       (draw-line left (+ top gap) left bottom)
-       (draw-line left bottom left (+ y height))
-       (draw-line right bottom right (+ y height)))
+           bottom (+ y (- height edge))
+           upper-trapezoid { 
+            :top-left [left y],
+            :top-right [right y],
+            :bottom-left [left top],
+            :bottom-right [right (- bottom gap)]
+           }
+           lower-trapezoid { 
+            :top-left [left (+ top gap)], 
+            :top-right [right bottom],
+            :bottom-left [left (+ y height)],
+            :bottom-right [right (+ y height)]
+           }]
+       (when fill [
+          (append-svg (svg/polygon (concat
+            (upper-trapezoid :top-left)
+            (upper-trapezoid :top-right)
+            (upper-trapezoid :bottom-right)
+            (upper-trapezoid :bottom-left)
+            ) :fill fill
+          ))
+          (append-svg (svg/polygon (concat
+            (lower-trapezoid :top-left)
+            (lower-trapezoid :top-right)
+            (lower-trapezoid :bottom-right)
+            (lower-trapezoid :bottom-left)
+            ) :fill fill
+          ))])
+       (apply draw-line (concat (upper-trapezoid :top-left)  (upper-trapezoid :bottom-left)))
```

Comment:
I don’t think there’s any need to use `apply` here, we can call `draw-line` directly with the results of looking up the values of the trapezoid corners. Also, it is a bit more idiomatic to use the keyword as the function, rather than the map, so:
```clojure
(draw-line (:top-left upper-trapezoid) (:bottom-left upper-trapezoid))
```

Reply:
I take this back! Trying to experiment with this refactoring I see that those corners are, of course, not single values, they are pairs of values, so `apply` is the only right way to do it. I would still swap the keywords and the map names in the lookups though, as a minor idiomatic point. Experienced clojure developers are used to reading keywords in function position as “look up this value in the map that follows”.

Reply:
(And again, it was clj-kondo immediately pointing out the folly of my efforts as I typed. 😄)

Reply:
No problem - it took me while to understand why I couldn't do this more simply the first time round too!